catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -270,8 +261,90 @@ def monitor_temperature(self):
             temperature = self.get_temperature()
             self.temperature.submit_data(np.array([temperature]))
 
+            if temperature > 30:
+                self.log.warning('Camera Temperature above 30 degrees, stopping acquisition and start fan')
+                self.fan_status = 'on'
+                self.should_shut_down = True
+
             self.sleep(0.1)
 
+    @property
+    def cooler_mode(self):
+        """
+        Check the cooling mode
+
+        Returns
+        -------
+        coolstr:
+            sensor cooler status 'on', 'off' or 'max'
+
+        """
+        coolint = self.cam.prop_getvalue(dcam.DCAM_IDPROP.SENSORCOOLER)
+
+        if coolint == 1.0:
+            return 'off'
+        elif coolint == 2.0:
+            return 'on'
+        else:
+            return 'max'
+
+    @cooler_mode.setter
+    def cooler_mode(self, coolstr):
+        """
+        Set the cooling mode
+
+        Parameterss
+        ----------
+        coolstr : str
+            Cooling mode of the camera
+        """
+
+        if coolstr == 'off':
+            coolint = 1.0
+        elif coolstr == 'on':
+            coolint = 2.0
+        elif coolstr == 'max':
+            coolint = 4.0
+        else:
+            raise ValueError('Water cooling mode not recognized.')
+
+        self.cam.prop_setvalue(dcam.DCAM_IDPROP.SENSORCOOLER, coolint)
+
+    @property
+    def fan_status(self):
+        """
+        Check fan status
+
+        Returns
+        -------
+        onoff: bool

```

Comment:
Should be type hint. Also might need spaces around the colon (:) for docstring rentering, I am unsure.